src/verifast.ml

```diff
@@ -714,7 +713,8 @@ module VerifyProgram(VerifyProgramArgs: VERIFY_PROGRAM_ARGS) = struct
                   let term = get_unique_var_symb pat tp' in
                   let term' =
                     match unfold_inferred_type tp with
-                      TypeParam x -> convert_provertype term (provertype_of_type tp') ProverInductive
+                      GhostTypeParam x -> convert_provertype term (provertype_of_type tp') ProverInductive
+                    | RealTypeParam x -> convert_provertype term (provertype_of_type tp') ProverInductive

```

Comment:
Nee, `RealTypeParam` is `ProverInt` en `ObjectType` ook, dus er is geen conversie nodig. Verwijder deze regel.